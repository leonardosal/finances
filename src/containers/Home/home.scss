.container {
  padding: 16px;

  @media (min-width: 1025px) {
    max-width: 1024px;
    margin: 0 auto;
  }
}

.box {
  background-color: #ccc;
  min-height: calc(100vh - 82px);
}

.button {
  padding: 8px;
}

.row {
  padding: 16px;
  border-bottom: 1px solid #eee;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 50px;
}
