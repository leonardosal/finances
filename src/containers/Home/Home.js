import React, { Component } from 'react';

import './home.scss';

export default class Home extends Component {
  addExpenses = () => {};

  render() {
    return (
      <div className="container">
        <button className="button" type="button">
          Adicionar
        </button>
        <div className="box">
          {[
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
            10,
            11,
            12,
            13,
            14,
            15,
            16,
            17,
            18,
            19,
            20,
            21,
            22,
            23,
            24,
            25,
            26,
          ].map((_, index) => (
            <div key={index + 1} className="row">
              <span>Almoço</span>
              <span>Alimentação</span>
              <span>R$ 100,00</span>
            </div>
          ))}
        </div>
      </div>
    );
  }
}
