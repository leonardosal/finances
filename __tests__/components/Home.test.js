import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../__mocks__/localStorage';
import Home from '../../src/containers/Home';

it('should render Home component without data saved', () => {
  const component = mount(<Home />);
  expect(toJson(component)).toMatchSnapshot();
});

it('should render Home component with data saved', () => {
  localStorage.setItem(
    '@transactions',
    JSON.stringify([
      { id: '0000-0000-0000', type: 'CREDIT', value: 1000.0 },
      { id: '0000-0000-0001', type: 'DEBIT', value: 100.0 },
      { id: '0000-0000-0002', type: 'DEBIT', value: 100.0 },
    ])
  );

  const component = mount(<Home />);
  expect(toJson(component)).toMatchSnapshot();
});

it('should render Home component when add transaction with valid data', () => {
  jest.mock('uuid', () => {
    return {
      v4: jest.fn(() => '0000-0000-0000'),
    };
  });
  const component = mount(<Home />);
  component.find('button#btn-open').simulate('click');
  component.find('select#type').simulate('change', {
    target: { value: 'DEBIT' },
  });
  component.find('input#value').simulate('change', {
    target: { value: 10 },
  });
  component.find('input#description').simulate('change', {
    target: { value: 'Uber' },
  });
  component.find('button#btn-save').simulate('click');
  component.find('form#transactionForm').simulate('submit');

  expect(toJson(component)).toMatchSnapshot();
});

it('should render Home component when add transaction with invalid data', () => {
  const component = mount(<Home />);
  component.find('button#btn-open').simulate('click');
  component.find('button#btn-save').simulate('click');
  component.find('form#transactionForm').simulate('submit');

  expect(toJson(component)).toMatchSnapshot();
});

it('should render Home component when add transaction and cancel', () => {
  const component = mount(<Home />);
  component.find('button#btn-open').simulate('click');
  component.find('button#btn-cancel').simulate('click');

  expect(toJson(component)).toMatchSnapshot();
});
