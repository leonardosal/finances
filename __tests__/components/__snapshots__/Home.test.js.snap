// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home component when add transaction and cancel 1`] = `
<Home>
  <div
    className="container"
  >
    <div
      className="box"
    >
      <CardInfo
        balance={790}
        handleClick={[Function]}
      >
        <div
          className="card-info"
        >
          <div>
            <h5>
              Saldo
            </h5>
            <h2>
              R$790,00
            </h2>
          </div>
          <Button
            id="btn-open"
            label="Adicionar Transação"
            onClick={[Function]}
            type="button"
          >
            <button
              className="button"
              id="btn-open"
              onClick={[Function]}
              type="button"
            >
              Adicionar Transação
            </button>
          </Button>
        </div>
      </CardInfo>
    </div>
    <div
      className="separator"
    />
    <div
      className="box"
    >
      <Table
        transactions={
          Array [
            Object {
              "description": "Uber",
              "id": "e281bc2d-1615-4db9-ae45-165b4dfff3db",
              "type": "DEBIT",
              "value": 10,
            },
            Object {
              "id": "0000-0000-0000",
              "type": "CREDIT",
              "value": 1000,
            },
            Object {
              "id": "0000-0000-0001",
              "type": "DEBIT",
              "value": 100,
            },
            Object {
              "id": "0000-0000-0002",
              "type": "DEBIT",
              "value": 100,
            },
          ]
        }
      >
        <Row
          description="Uber"
          key="e281bc2d-1615-4db9-ae45-165b4dfff3db"
          type="DEBIT"
          value={10}
        >
          <div
            className="row"
          >
            <span>
              Uber
            </span>
            <span
              className="debit"
            >
              R$10,00
            </span>
          </div>
        </Row>
        <Row
          description=""
          key="0000-0000-0000"
          type="CREDIT"
          value={1000}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="credit"
            >
              R$1.000,00
            </span>
          </div>
        </Row>
        <Row
          description=""
          key="0000-0000-0001"
          type="DEBIT"
          value={100}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="debit"
            >
              R$100,00
            </span>
          </div>
        </Row>
        <Row
          description=""
          key="0000-0000-0002"
          type="DEBIT"
          value={100}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="debit"
            >
              R$100,00
            </span>
          </div>
        </Row>
      </Table>
    </div>
  </div>
  <Modal
    onChangeField={[Function]}
    show={false}
    submitForm={[Function]}
    toggleModal={[Function]}
    transaction={
      Object {
        "description": "",
        "type": "DEBIT",
        "value": 0,
      }
    }
  >
    <div
      className="modal"
      id="modal"
    >
      <div
        className="modal-content"
      >
        <h3>
          Adicionar transação
        </h3>
        <form
          className="form"
          id="transactionForm"
          name="transactionForm"
          onSubmit={[Function]}
        >
          <label
            htmlFor="type"
          >
            Tipo
            <select
              id="type"
              name="type"
              onChange={[Function]}
              value="DEBIT"
            >
              <option
                value="DEBIT"
              >
                Despesa
              </option>
              <option
                value="CREDIT"
              >
                Receita
              </option>
            </select>
          </label>
          <label
            htmlFor="value"
          >
            Valor
            <input
              id="value"
              name="value"
              onChange={[Function]}
              type="number"
              value={0}
            />
          </label>
          <label
            htmlFor="description"
          >
            Descrição
            <input
              id="description"
              name="description"
              onChange={[Function]}
              value=""
            />
          </label>
          <div
            className="footer"
          >
            <Button
              id="btn-save"
              label="Salvar"
              onClick={[Function]}
              type="submit"
            >
              <button
                className="button"
                id="btn-save"
                onClick={[Function]}
                type="submit"
              >
                Salvar
              </button>
            </Button>
            <Button
              id="btn-cancel"
              label="Cancelar"
              onClick={[Function]}
              type="button"
            >
              <button
                className="button"
                id="btn-cancel"
                onClick={[Function]}
                type="button"
              >
                Cancelar
              </button>
            </Button>
          </div>
        </form>
      </div>
    </div>
  </Modal>
</Home>
`;

exports[`should render Home component when add transaction with invalid data 1`] = `
<Home>
  <div
    className="container"
  >
    <div
      className="box"
    >
      <CardInfo
        balance={790}
        handleClick={[Function]}
      >
        <div
          className="card-info"
        >
          <div>
            <h5>
              Saldo
            </h5>
            <h2>
              R$790,00
            </h2>
          </div>
          <Button
            id="btn-open"
            label="Adicionar Transação"
            onClick={[Function]}
            type="button"
          >
            <button
              className="button"
              id="btn-open"
              onClick={[Function]}
              type="button"
            >
              Adicionar Transação
            </button>
          </Button>
        </div>
      </CardInfo>
    </div>
    <div
      className="separator"
    />
    <div
      className="box"
    >
      <Table
        transactions={
          Array [
            Object {
              "description": "Uber",
              "id": "e281bc2d-1615-4db9-ae45-165b4dfff3db",
              "type": "DEBIT",
              "value": 10,
            },
            Object {
              "id": "0000-0000-0000",
              "type": "CREDIT",
              "value": 1000,
            },
            Object {
              "id": "0000-0000-0001",
              "type": "DEBIT",
              "value": 100,
            },
            Object {
              "id": "0000-0000-0002",
              "type": "DEBIT",
              "value": 100,
            },
          ]
        }
      >
        <Row
          description="Uber"
          key="e281bc2d-1615-4db9-ae45-165b4dfff3db"
          type="DEBIT"
          value={10}
        >
          <div
            className="row"
          >
            <span>
              Uber
            </span>
            <span
              className="debit"
            >
              R$10,00
            </span>
          </div>
        </Row>
        <Row
          description=""
          key="0000-0000-0000"
          type="CREDIT"
          value={1000}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="credit"
            >
              R$1.000,00
            </span>
          </div>
        </Row>
        <Row
          description=""
          key="0000-0000-0001"
          type="DEBIT"
          value={100}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="debit"
            >
              R$100,00
            </span>
          </div>
        </Row>
        <Row
          description=""
          key="0000-0000-0002"
          type="DEBIT"
          value={100}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="debit"
            >
              R$100,00
            </span>
          </div>
        </Row>
      </Table>
    </div>
  </div>
  <Modal
    onChangeField={[Function]}
    show={true}
    submitForm={[Function]}
    toggleModal={[Function]}
    transaction={
      Object {
        "description": "",
        "type": "DEBIT",
        "value": 0,
      }
    }
  >
    <div
      className="modal show"
      id="modal"
    >
      <div
        className="modal-content"
      >
        <h3>
          Adicionar transação
        </h3>
        <form
          className="form"
          id="transactionForm"
          name="transactionForm"
          onSubmit={[Function]}
        >
          <label
            htmlFor="type"
          >
            Tipo
            <select
              id="type"
              name="type"
              onChange={[Function]}
              value="DEBIT"
            >
              <option
                value="DEBIT"
              >
                Despesa
              </option>
              <option
                value="CREDIT"
              >
                Receita
              </option>
            </select>
          </label>
          <label
            htmlFor="value"
          >
            Valor
            <input
              id="value"
              name="value"
              onChange={[Function]}
              type="number"
              value={0}
            />
          </label>
          <label
            htmlFor="description"
          >
            Descrição
            <input
              id="description"
              name="description"
              onChange={[Function]}
              value=""
            />
          </label>
          <div
            className="footer"
          >
            <Button
              id="btn-save"
              label="Salvar"
              onClick={[Function]}
              type="submit"
            >
              <button
                className="button"
                id="btn-save"
                onClick={[Function]}
                type="submit"
              >
                Salvar
              </button>
            </Button>
            <Button
              id="btn-cancel"
              label="Cancelar"
              onClick={[Function]}
              type="button"
            >
              <button
                className="button"
                id="btn-cancel"
                onClick={[Function]}
                type="button"
              >
                Cancelar
              </button>
            </Button>
          </div>
        </form>
      </div>
    </div>
  </Modal>
</Home>
`;

exports[`should render Home component when add transaction with valid data 1`] = `
<Home>
  <div
    className="container"
  >
    <div
      className="box"
    >
      <CardInfo
        balance={790}
        handleClick={[Function]}
      >
        <div
          className="card-info"
        >
          <div>
            <h5>
              Saldo
            </h5>
            <h2>
              R$790,00
            </h2>
          </div>
          <Button
            id="btn-open"
            label="Adicionar Transação"
            onClick={[Function]}
            type="button"
          >
            <button
              className="button"
              id="btn-open"
              onClick={[Function]}
              type="button"
            >
              Adicionar Transação
            </button>
          </Button>
        </div>
      </CardInfo>
    </div>
    <div
      className="separator"
    />
    <div
      className="box"
    >
      <Table
        transactions={
          Array [
            Object {
              "description": "Uber",
              "id": "e281bc2d-1615-4db9-ae45-165b4dfff3db",
              "type": "DEBIT",
              "value": 10,
            },
            Object {
              "id": "0000-0000-0000",
              "type": "CREDIT",
              "value": 1000,
            },
            Object {
              "id": "0000-0000-0001",
              "type": "DEBIT",
              "value": 100,
            },
            Object {
              "id": "0000-0000-0002",
              "type": "DEBIT",
              "value": 100,
            },
          ]
        }
      >
        <Row
          description="Uber"
          key="e281bc2d-1615-4db9-ae45-165b4dfff3db"
          type="DEBIT"
          value={10}
        >
          <div
            className="row"
          >
            <span>
              Uber
            </span>
            <span
              className="debit"
            >
              R$10,00
            </span>
          </div>
        </Row>
        <Row
          description=""
          key="0000-0000-0000"
          type="CREDIT"
          value={1000}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="credit"
            >
              R$1.000,00
            </span>
          </div>
        </Row>
        <Row
          description=""
          key="0000-0000-0001"
          type="DEBIT"
          value={100}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="debit"
            >
              R$100,00
            </span>
          </div>
        </Row>
        <Row
          description=""
          key="0000-0000-0002"
          type="DEBIT"
          value={100}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="debit"
            >
              R$100,00
            </span>
          </div>
        </Row>
      </Table>
    </div>
  </div>
  <Modal
    onChangeField={[Function]}
    show={false}
    submitForm={[Function]}
    toggleModal={[Function]}
    transaction={
      Object {
        "description": "",
        "type": "DEBIT",
        "value": 0,
      }
    }
  >
    <div
      className="modal"
      id="modal"
    >
      <div
        className="modal-content"
      >
        <h3>
          Adicionar transação
        </h3>
        <form
          className="form"
          id="transactionForm"
          name="transactionForm"
          onSubmit={[Function]}
        >
          <label
            htmlFor="type"
          >
            Tipo
            <select
              id="type"
              name="type"
              onChange={[Function]}
              value="DEBIT"
            >
              <option
                value="DEBIT"
              >
                Despesa
              </option>
              <option
                value="CREDIT"
              >
                Receita
              </option>
            </select>
          </label>
          <label
            htmlFor="value"
          >
            Valor
            <input
              id="value"
              name="value"
              onChange={[Function]}
              type="number"
              value={0}
            />
          </label>
          <label
            htmlFor="description"
          >
            Descrição
            <input
              id="description"
              name="description"
              onChange={[Function]}
              value=""
            />
          </label>
          <div
            className="footer"
          >
            <Button
              id="btn-save"
              label="Salvar"
              onClick={[Function]}
              type="submit"
            >
              <button
                className="button"
                id="btn-save"
                onClick={[Function]}
                type="submit"
              >
                Salvar
              </button>
            </Button>
            <Button
              id="btn-cancel"
              label="Cancelar"
              onClick={[Function]}
              type="button"
            >
              <button
                className="button"
                id="btn-cancel"
                onClick={[Function]}
                type="button"
              >
                Cancelar
              </button>
            </Button>
          </div>
        </form>
      </div>
    </div>
  </Modal>
</Home>
`;

exports[`should render Home component with data saved 1`] = `
<Home>
  <div
    className="container"
  >
    <div
      className="box"
    >
      <CardInfo
        balance={800}
        handleClick={[Function]}
      >
        <div
          className="card-info"
        >
          <div>
            <h5>
              Saldo
            </h5>
            <h2>
              R$800,00
            </h2>
          </div>
          <Button
            id="btn-open"
            label="Adicionar Transação"
            onClick={[Function]}
            type="button"
          >
            <button
              className="button"
              id="btn-open"
              onClick={[Function]}
              type="button"
            >
              Adicionar Transação
            </button>
          </Button>
        </div>
      </CardInfo>
    </div>
    <div
      className="separator"
    />
    <div
      className="box"
    >
      <Table
        transactions={
          Array [
            Object {
              "id": "0000-0000-0000",
              "type": "CREDIT",
              "value": 1000,
            },
            Object {
              "id": "0000-0000-0001",
              "type": "DEBIT",
              "value": 100,
            },
            Object {
              "id": "0000-0000-0002",
              "type": "DEBIT",
              "value": 100,
            },
          ]
        }
      >
        <Row
          description=""
          key="0000-0000-0000"
          type="CREDIT"
          value={1000}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="credit"
            >
              R$1.000,00
            </span>
          </div>
        </Row>
        <Row
          description=""
          key="0000-0000-0001"
          type="DEBIT"
          value={100}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="debit"
            >
              R$100,00
            </span>
          </div>
        </Row>
        <Row
          description=""
          key="0000-0000-0002"
          type="DEBIT"
          value={100}
        >
          <div
            className="row"
          >
            <span />
            <span
              className="debit"
            >
              R$100,00
            </span>
          </div>
        </Row>
      </Table>
    </div>
  </div>
  <Modal
    onChangeField={[Function]}
    show={false}
    submitForm={[Function]}
    toggleModal={[Function]}
    transaction={
      Object {
        "description": "",
        "type": "DEBIT",
        "value": 0,
      }
    }
  >
    <div
      className="modal"
      id="modal"
    >
      <div
        className="modal-content"
      >
        <h3>
          Adicionar transação
        </h3>
        <form
          className="form"
          id="transactionForm"
          name="transactionForm"
          onSubmit={[Function]}
        >
          <label
            htmlFor="type"
          >
            Tipo
            <select
              id="type"
              name="type"
              onChange={[Function]}
              value="DEBIT"
            >
              <option
                value="DEBIT"
              >
                Despesa
              </option>
              <option
                value="CREDIT"
              >
                Receita
              </option>
            </select>
          </label>
          <label
            htmlFor="value"
          >
            Valor
            <input
              id="value"
              name="value"
              onChange={[Function]}
              type="number"
              value={0}
            />
          </label>
          <label
            htmlFor="description"
          >
            Descrição
            <input
              id="description"
              name="description"
              onChange={[Function]}
              value=""
            />
          </label>
          <div
            className="footer"
          >
            <Button
              id="btn-save"
              label="Salvar"
              onClick={[Function]}
              type="submit"
            >
              <button
                className="button"
                id="btn-save"
                onClick={[Function]}
                type="submit"
              >
                Salvar
              </button>
            </Button>
            <Button
              id="btn-cancel"
              label="Cancelar"
              onClick={[Function]}
              type="button"
            >
              <button
                className="button"
                id="btn-cancel"
                onClick={[Function]}
                type="button"
              >
                Cancelar
              </button>
            </Button>
          </div>
        </form>
      </div>
    </div>
  </Modal>
</Home>
`;

exports[`should render Home component without data saved 1`] = `
<Home>
  <div
    className="container"
  >
    <div
      className="box"
    >
      <CardInfo
        balance={0}
        handleClick={[Function]}
      >
        <div
          className="card-info"
        >
          <div>
            <h5>
              Saldo
            </h5>
            <h2>
              R$0,00
            </h2>
          </div>
          <Button
            id="btn-open"
            label="Adicionar Transação"
            onClick={[Function]}
            type="button"
          >
            <button
              className="button"
              id="btn-open"
              onClick={[Function]}
              type="button"
            >
              Adicionar Transação
            </button>
          </Button>
        </div>
      </CardInfo>
    </div>
    <div
      className="separator"
    />
    <div
      className="box"
    >
      <Table
        transactions={Array []}
      />
    </div>
  </div>
  <Modal
    onChangeField={[Function]}
    show={false}
    submitForm={[Function]}
    toggleModal={[Function]}
    transaction={
      Object {
        "description": "",
        "type": "DEBIT",
        "value": 0,
      }
    }
  >
    <div
      className="modal"
      id="modal"
    >
      <div
        className="modal-content"
      >
        <h3>
          Adicionar transação
        </h3>
        <form
          className="form"
          id="transactionForm"
          name="transactionForm"
          onSubmit={[Function]}
        >
          <label
            htmlFor="type"
          >
            Tipo
            <select
              id="type"
              name="type"
              onChange={[Function]}
              value="DEBIT"
            >
              <option
                value="DEBIT"
              >
                Despesa
              </option>
              <option
                value="CREDIT"
              >
                Receita
              </option>
            </select>
          </label>
          <label
            htmlFor="value"
          >
            Valor
            <input
              id="value"
              name="value"
              onChange={[Function]}
              type="number"
              value={0}
            />
          </label>
          <label
            htmlFor="description"
          >
            Descrição
            <input
              id="description"
              name="description"
              onChange={[Function]}
              value=""
            />
          </label>
          <div
            className="footer"
          >
            <Button
              id="btn-save"
              label="Salvar"
              onClick={[Function]}
              type="submit"
            >
              <button
                className="button"
                id="btn-save"
                onClick={[Function]}
                type="submit"
              >
                Salvar
              </button>
            </Button>
            <Button
              id="btn-cancel"
              label="Cancelar"
              onClick={[Function]}
              type="button"
            >
              <button
                className="button"
                id="btn-cancel"
                onClick={[Function]}
                type="button"
              >
                Cancelar
              </button>
            </Button>
          </div>
        </form>
      </div>
    </div>
  </Modal>
</Home>
`;
